packages/auth/src/signature/middleware.test.ts

```diff
@@ -401,6 +401,101 @@ describe('Signature Service', (): void => {
       scope.isDone()
     })
 
+    test('middleware succeeds if BYPASS_SIGNATURE_VALIDATION is true with bad signature', async (): Promise<void> => {
+      const defaultByPassSignatureValidation = Config.bypassSignatureValidation
+      Config.bypassSignatureValidation = true
+
+      const scope = nock(KEY_REGISTRY_ORIGIN)
+        .get(keyPath)
+        .reply(200, {
+          jwk: testClientKey.jwk,
+          client: TEST_CLIENT
+        } as ClientKey)
+
+      const ctx = await createContextWithSigHeaders(
+        {
+          headers: {
+            Accept: 'application/json'
+          },
+          url: '/',
+          method: 'POST'
+        },
+        {},
+        {
+          client: {
+            display: TEST_CLIENT_DISPLAY,
+            key: {
+              proof: 'httpsig',
+              jwk: testClientKey.jwk
+            }
+          }
+        },
+        privateKey,
+        deps
+      )
+
+      ctx.headers['signature'] = 'wrong-signature'
+
+      await grantInitiationHttpsigMiddleware(ctx, next)
+
+      expect(ctx.response.status).toEqual(200)
+      expect(next).toHaveBeenCalled()
+
+      Config.bypassSignatureValidation = defaultByPassSignatureValidation
+
+      scope.isDone()
```

Comment:
```suggestion
      scope.done()
```
- https://github.com/interledger/rafiki/issues/656